earn/src/data/LendingPair.ts

```diff
@@ -77,79 +80,153 @@ export async function getAvailableLendingPairs(
     };
   });
 
-  const kittyLens = new ethers.Contract(ALOE_II_KITTY_LENS_ADDRESS, KittyLensABI, provider);
-  const oracle = new ethers.Contract(ALOE_II_ORACLE, VolatilityOracleABI, provider);
-
-  const unfilteredPairs: Array<LendingPair | null> = await Promise.all(
-    addresses.map(async (market) => {
-      const uniswapPool = new ethers.Contract(market.pool, UniswapV3PoolABI, provider);
-
-      const [basics0, basics1, feeTier, oracleResult] = await Promise.all([
-        kittyLens.readBasics(market.kitty0),
-        kittyLens.readBasics(market.kitty1),
-        uniswapPool.fee(),
-        oracle.consult(market.pool),
-      ]);
-
-      const token0 = getToken(chain.id, basics0.asset);
-      const token1 = getToken(chain.id, basics1.asset);
-      if (token0 == null || token1 == null) return null;
-      const kitty0 = new Kitty(
-        chain.id,
-        market.kitty0 as Address,
-        token0.decimals,
-        `${token0.ticker}+`,
-        `Aloe II ${token0.name}`,
-        token0.iconPath,
-        token0
-      );
-      const kitty1 = new Kitty(
-        chain.id,
-        market.kitty1 as Address,
-        token1.decimals,
-        `${token1.ticker}+`,
-        `Aloe II ${token1.name}`,
-        token1.iconPath,
-        token1
-      );
-
-      const utilization0 = new Big(basics0.utilization.toString()).div(10 ** 18).toNumber();
-      const utilization1 = new Big(basics1.utilization.toString()).div(10 ** 18).toNumber();
-
-      const interestRate0 = new Big(basics0.interestRate.toString());
-      const interestRate1 = new Big(basics1.interestRate.toString());
-      // SupplyAPY = Utilization * (1 - reservePercentage) * BorrowAPY
-      const APY0 = utilization0 * (1 - 1 / 8) * (interestRate0.div(10 ** 12).toNumber() ** (365 * 24 * 60 * 60) - 1.0);
-      const APY1 = utilization1 * (1 - 1 / 8) * (interestRate1.div(10 ** 12).toNumber() ** (365 * 24 * 60 * 60) - 1.0);
-
-      let IV = oracleResult[1].div(1e9).toNumber() / 1e9;
-      // Annualize it
-      IV *= Math.sqrt(365);
-
-      return new LendingPair(
+  const contractCallContexts: ContractCallContext[] = [];
+
+  addresses.forEach((market) => {
```

Comment:
At some point this has got to break right? I can't imagine we can stuff 100 different lender requests into a single multicall, for example. Maybe worthwhile to paginate this somehow?

Reply:
I just load-tested it and was able to do 6750 total calls without it failing. With that said, each lender does require more than one call, but that should still give us plenty of breathing room.